src/Sylius/Component/Core/Provider/ChannelBasedLocaleProvider.php

```diff
@@ -54,6 +54,24 @@ public function getAvailableLocalesCodes(): array
         }
     }
 
+    public function isLocaleCodeAvailable(string $locale): bool
+    {
+        try {
+            /** @var ChannelInterface $channel */
+            $channel = $this->channelContext->getChannel();
+
+            return $channel
+                ->getLocales()
+                ->map(function (LocaleInterface $locale) {
```

Comment:
You could use `->exists()` instead of `->map()`, like:
```
$channel
    ->getLocales()
    ->exists(static function (LocaleInterface $locale) use ($localeCode) {
        return $locale->getCode() === $localeCode;
    })
```

Reply:
Nice, haven't thought about that.